### ISSUE TITLE
generate shapes during validation in training

### ISSUE THREAD

**Issue by haozhaopop** [HobbitLong/shape2prog#4]
In train_program_generator.py, during validation, if we set gen_shape=True, we actually render shapes on step level, rather than block level, I am wondering if this is a bug?

**Comment by HobbitLong** [comment]
Good catch! You are right, @haozhaopop . Please either follow `test.py` to render shapes or calling for `render_block` function.
